deploy.sh

```diff
@@ -49,47 +56,53 @@ cd "$SVN_DIR"
 svn update --set-depth infinity assets
 svn update --set-depth infinity trunk
 
-echo "➤ Copying files..."
-if [[ -e "$GITHUB_WORKSPACE/.distignore" ]]; then
-	echo "ℹ︎ Using .distignore"
-	# Copy from current branch to /trunk, excluding dotorg assets
-	# The --delete flag will delete anything in destination that no longer exists in source
-	rsync -rc --exclude-from="$GITHUB_WORKSPACE/.distignore" "$GITHUB_WORKSPACE/" trunk/ --delete --delete-excluded
-else
-	echo "ℹ︎ Using .gitattributes"
-
-	cd "$GITHUB_WORKSPACE"
-
-	# "Export" a cleaned copy to a temp directory
-	TMP_DIR="${HOME}/archivetmp"
-	mkdir "$TMP_DIR"
-
-	git config --global user.email "10upbot+github@10up.com"
-	git config --global user.name "10upbot on GitHub"
-
-	# If there's no .gitattributes file, write a default one into place
-	if [[ ! -e "$GITHUB_WORKSPACE/.gitattributes" ]]; then
-		cat > "$GITHUB_WORKSPACE/.gitattributes" <<-EOL
-		/$ASSETS_DIR export-ignore
-		/.gitattributes export-ignore
-		/.gitignore export-ignore
-		/.github export-ignore
-		EOL
-
-		# Ensure we are in the $GITHUB_WORKSPACE directory, just in case
-		# The .gitattributes file has to be committed to be used
-		# Just don't push it to the origin repo :)
-		git add .gitattributes && git commit -m "Add .gitattributes file"
-	fi
-
-	# This will exclude everything in the .gitattributes file with the export-ignore flag
-	git archive HEAD | tar x --directory="$TMP_DIR"
-
-	cd "$SVN_DIR"
 
-	# Copy from clean copy to /trunk, excluding dotorg assets
-	# The --delete flag will delete anything in destination that no longer exists in source
-	rsync -rc "$TMP_DIR/" trunk/ --delete --delete-excluded
+if [[ "$BUILD_DIR" = false ]]; then
+	echo "➤ Copying files..."
+	if [[ -e "$GITHUB_WORKSPACE/.distignore" ]]; then
+		echo "ℹ︎ Using .distignore"
+		# Copy from current branch to /trunk, excluding dotorg assets
+		# The --delete flag will delete anything in destination that no longer exists in source
+		rsync -rc --exclude-from="$GITHUB_WORKSPACE/.distignore" "$GITHUB_WORKSPACE/" trunk/ --delete --delete-excluded
+	else
+		echo "ℹ︎ Using .gitattributes"
+
+		cd "$GITHUB_WORKSPACE"
+
+		# "Export" a cleaned copy to a temp directory
+		TMP_DIR="${HOME}/archivetmp"
+		mkdir "$TMP_DIR"
+
+		git config --global user.email "10upbot+github@10up.com"
+		git config --global user.name "10upbot on GitHub"
+
+		# If there's no .gitattributes file, write a default one into place
+		if [[ ! -e "$GITHUB_WORKSPACE/.gitattributes" ]]; then
+			cat > "$GITHUB_WORKSPACE/.gitattributes" <<-EOL
+			/$ASSETS_DIR export-ignore
+			/.gitattributes export-ignore
+			/.gitignore export-ignore
+			/.github export-ignore
+			EOL
+
+			# Ensure we are in the $GITHUB_WORKSPACE directory, just in case
+			# The .gitattributes file has to be committed to be used
+			# Just don't push it to the origin repo :)
+			git add .gitattributes && git commit -m "Add .gitattributes file"
+		fi
```

Comment:
@peterwilsoncc I decided not to use `.disignore` or `.gitattributes` because I assume that if plugin authors use `BUILD_DIR`, they are using a command/script to build the plugin, so anything inside `BUILD_DIR` is meant to be deployed. They have control over any files copied to that folder. Using `.disignore` or `.gitattributes` here can cause confusions IMO.